tests/import_test.go

```diff
@@ -184,6 +186,110 @@ var _ = Describe("[rfe_id:1115][crit:high][vendor:cnv-qe@redhat.com][level:compo
 
 })
 
+var _ = Describe("DataVolume Garbage Collection", func() {
+	var (
+		f        = framework.NewFramework(namespacePrefix)
+		ns       string
+		err      error
+		config   *cdiv1.CDIConfig
+		origSpec *cdiv1.CDIConfigSpec
+	)
+
+	BeforeEach(func() {
+		ns = f.Namespace.Name
+		config, err = f.CdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		origSpec = config.Spec.DeepCopy()
+	})
+
+	AfterEach(func() {
+		By("Restoring CDIConfig to original state")
+		err = utils.UpdateCDIConfig(f.CrClient, func(config *cdiv1.CDIConfigSpec) {
+			origSpec.DeepCopyInto(config)
+		})
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() bool {
+			config, err = f.CdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return reflect.DeepEqual(config.Spec, *origSpec)
+		}, 30*time.Second, time.Second).Should(BeTrue())
+	})
+
+	verifyGC := func(dvName string) {
+		By("Wait for import to be completed")
+		err = utils.WaitForDataVolumePhase(f, ns, cdiv1.Succeeded, dvName)
+		Expect(err).ToNot(HaveOccurred(), "Datavolume not in phase succeeded in time")
+
+		By("Wait for DV to be garbage collected")
+		Eventually(func() bool {
+			_, err = f.CdiClient.CdiV1beta1().DataVolumes(ns).Get(context.TODO(), dvName, metav1.GetOptions{})
+			return k8serrors.IsNotFound(err)
+		}, timeout, pollingInterval).Should(BeTrue())
+
+		By("Verify PVC still exists")
+		pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(ns).Get(context.TODO(), dvName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(controller.IsSucceeded(pvc)).To(BeTrue())
+		Expect(pvc.Annotations[controller.AnnDeleteAfterCompletion]).To(Equal("true"))
+
+		By("Verify PVC gets DV original OwnerReferences, and the DV reference is removed")
+		Expect(pvc.OwnerReferences).Should(HaveLen(1))
+		Expect(pvc.OwnerReferences[0].UID).Should(Equal(config.UID))
+	}
+
+	verifyNoGC := func(dvName string) {
+		By("Verify DV is not garbage collected")
+		matchString := "DataVolume is not garbage collected per annotation\t{\"Datavolume\": \"" + ns + "/" + dvName + "\"}"
+		fmt.Fprintf(GinkgoWriter, "INFO: matchString: [%s]\n", matchString)
+		Eventually(func() string {
+			log, err := tests.RunKubectlCommand(f, "logs", f.ControllerPod.Name, "-n", f.CdiInstallNs)
+			Expect(err).NotTo(HaveOccurred())
+			return log
+		}, controllerSkipPVCCompleteTimeout, assertionPollInterval).Should(ContainSubstring(matchString))
+		Expect(err).ToNot(HaveOccurred())
+		dv, err := f.CdiClient.CdiV1beta1().DataVolumes(ns).Get(context.TODO(), dvName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(dv.Annotations[controller.AnnDeleteAfterCompletion]).To(Equal("false"))
+	}
+
+	DescribeTable("Should", func(annDeleteAfterCompletion string, dvFunc func(dvName string)) {
```

Comment:
So dvFunc doesn't really tell me what this function does, maybe name it verifyGCFunc that tells me we are verifying something about garbage collection.